### DISCUSSION TITLE
Issue uploading custom Tasmota firmware

### DISCUSSION THREAD

**Discussion by danielvega3009** [arendst/Tasmota#12703] [Q&A] [answered]
Hi, 

I have modified the tasmota firmware to include a new functionality for a personal project, when uploading the firmware to my tasmota device from the application, it gives me the following error:

![image](https://user-images.githubusercontent.com/84938046/126333324-0068773d-1b33-42cf-b2cb-d7f566dff1a5.png)

How could I flash my tasmota device?? without welding please.

**Comment by ascillato** [comment, marked as answer]
https://github.com/arendst/Tasmota/discussions/12286#discussioncomment-975277


### If you see the message **_"upload buffer miscompare"_** means not enough space. Please, use minimal or lite and then use the full version you want.

**Reply by danielvega3009** [reply]
Hi, thanks
I don't understand about minimal or lite, How can I do it?

**Reply by sfromis** [reply]
It is a smaller binary which you can "upgrade" to first, to do your own upgrade next. And always, always use the .gz binaries (not the .bin ones) for OTA upgrading. Precompiled binaries here http://ota.tasmota.com/tasmota/

**Reply by ascillato** [reply]
> I don't understand about minimal or lite, How can I do it?

@danielvega3009

Hi, as explained in the summary (in the link provided), you need to free some space in your flash.

So:

1 -  Upgrade to http://ota.tasmota.com/tasmota/tasmota-minimal.bin.gz This will free up space.
2 - Upgrade to your FULL version.

**Reply by danielvega3009** [reply]
Hi again,

I did what you said, first I uploaded the lite version and then mine, the problem is that after uploading my firmware I noticed that now the tasmota application was very slow, that is, I clicked on settings for example and it took almost a minute to open, so I decided to go back to the lite version. During the upload of the lite version the screen went blank, as if I had lost the connection and now it seems that the tasmota device does not work, because it does not appear on the network, any solution? I have tried to reset it by pressing the device button for more than 10 seconds and nothing

**Reply by barbudor** [reply]
Recovery requires Button_1 to be pressed more than 40 seconds and of course it needs a GPIO to be configured as Button_1. It won't work on a brand new device that has never been configured.

What should work in any case is power cycling the device 7 times with 2 seconds off, 2 seconds on. Power cycling do not mean cycling the relay controlled by the device, it means actually cutting power to the device itself. 

I see that you never elaborate on what device you are working on. Any chance to be able to use serial?

**Reply by ascillato** [reply]
> ... the problem is that after uploading my firmware I noticed that now the tasmota application was very slow ...

I'm sorry but as it is your own custom version, we don't know what you have changed hence we can not help you. If you have used delays or blocked loops, your modified version won't work. So, about upgrading, we can help, but why your own firmware does not work, we can't as we don't have it. Sorry

**Reply by danielvega3009** [reply]
my custom version is exactly like the last version of Tasmota in Github, but I added a new function that when the device connect to wifi, the device does a query a database to check if the device is registered in the network, because I have a local network where every device sign up automatically in a data base, only that, this code was added in the xdrv_01_webserver.ino file  into the StartWebserver function

**Reply by danielvega3009** [reply]
@barbudor  my device is athom smart plug based in tasmota https://es.athom.tech/blank-1/color-smart-plug

**Reply by ascillato2** [reply]
About upgrading, we can help, but why your own firmware does not work, we can't as we don't have it. Sorry.

**Reply by barbudor** [reply]
Your change may seems small to you eyes but it may impact Tasmota event loop.
I suggest restarting from original code and adding pieces of code one by one to idendity which piece creates the problem
